src/ebmc/bdd_engine.cpp

```diff
@@ -455,6 +504,12 @@ void bdd_enginet::check_property(propertyt &property)
   if(property.is_assumed())
     return;
 
+  if(!property_supported(property.normalized_expr))
+  {
+    property.failure("property not supported by BDD engine");
+    return;
+  }
+
```

Comment:
Should the code that follows (the code that actually does the transform to CTL) be combined with the `property_supported` code? That is, should `property_supported` right away do the transform?

Reply:
The duplication is also bugging me, but it seems even less desirable to make the creation of some new data structure a side-effect of `property_supported`.

Reply:
How about making `property_supported` (though then perhaps by the name of `make_CTL_property` or the likes) return a `std::optional<exprt>` where an unsupported property would cause `{}` to be returned?

Reply:
Coming with #1051